akka-typed/src/main/scala/akka/typed/Deployment.scala

```diff
@@ -10,22 +10,62 @@ import scala.annotation.tailrec
 
 /**
  * Data structure for describing an actor’s deployment details like which
- * executor to run it on.
+ * executor to run it on. For each type of setting (e.g. [[DispatcherSelector]]
+ * or [[MailboxCapacity]]) the FIRST occurrence is used when creating the
+ * actor; this means that adding configuration using the "with" methods
+ * overrides what was configured previously.
  *
  * Deliberately not sealed in order to emphasize future extensibility by the
  * framework—this is not intended to be extended by user code.
+ *
+ * The DeploymentConfig includes a `next` reference so that it can form an
+ * internally linked list. Traversal of this list stops when encountering the
+ * [[EmptyDeploymentConfig$]] object.
  */
 abstract class DeploymentConfig extends Product with Serializable {
```

Comment:
When computing the LUB of some case class instances below, the compile will find that they are subtypes of `DeploymentConfig with Product with Serializable` unless we fold these into the base type—I’ve seen this “trick” (or simplification) at scala.world and thought I’d use it.

Unfortunately I didn’t read my own comment about extensibility, meaning that here it is actually wrong because we cannot assume that all subtypes will be case classes, someone might use it from Java as well and do things.

In this vein we might want to discuss whether an actor’s DeploymentConfig should be retained by the ActorCell such that it can be obtained from the ActorContext. This would allow user code to pass parameters in this fashion as well, but I’m uncertain as to whether that is a good idea.